cpp/arcticdb/async/task_scheduler.hpp

```diff
@@ -66,7 +66,14 @@ class InstrumentedNamedFactory : public folly::ThreadFactory{
                 ARCTICDB_SAMPLE_THREAD();
               func();
             });
+    
   }
+// we use a modern version of folly when consuming dependencies from conda
```

Comment:
I think we should use the more recent versions for conda.  There won't be a windows nor Apple silicon build on conda for the old version used.
Also on conda-forge itself, with the global pinnings (https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml), using the old version of `folly` leads to unsatisfiable environments.

Reply:
Then the folly in the non-Conda (vcpkg) build should be updated to match. Otherwise, there could be mysterious bugs that only affect one particular build.

Reply:
I think we _have_ to use a more recent version of Folly. 

Rather than downgrading Conda, we should upgrade PyPi IMO. This can't block us for now. Keep in mind that there can't be anything fundamentally broken as the tests have passed.

Reply:
Due to the cost of changing a version of folly (e.g. resolving errors due to warnings being promoted the varying strictness of compilers across platform took us days before eventually converging on this configuration), the small intersection of versions of folly on conda-forge and vcpkg, the absence of test failures, and retro compatibility of this new version, I think it is reasonable using this more recent version of folly in the meantime. Still, I think it makes sense to use the exact same versions of dependencies as much as possible on the long term. 

I am unknowledgeable about your setup using vcpkg and about vcpkg generally, but it seems that vcpkg not to have kept with the last versions of folly ([its browser](https://vcpkg.io/en/packages.html) shows folly's last version is `2022.08.15.00`), while [conda-forge has been releasing all the last versions](https://anaconda.org/conda-forge/folly). How should we address this situation?

Reply:
In a general case, having obvious version-dependent handling in code (like this `#ifdef ARCTICDB_USING_CONDA`) is not a good sync of library backward compatibility.

However, I can accept this specific case as a temporary solution, but please log an issue to unify the code.

The vcpkg web interface is not very reliable. You can see version list here: https://github.com/microsoft/vcpkg/blob/master/versions/f-/folly.json

Reply:
I agree that using constant definition is definitely bad practice, but we might need to not rely on submodules first. 

I have opened https://github.com/man-group/ArcticDB/issues/422 in this regard.